BillOrSteve/ViewController.swift

```diff
@@ -10,16 +10,114 @@ import UIKit
+                "In 1994, he was asked by a TV interviewer if he could jump over a chair from a standing position. He promptly took the challenge and leapt over the chair like a boss."
+            ]
+            let steveFacts: [String] = [
+                "He took a calligraphy course, which he says was instrumental in the future company products' attention to typography and font.",
+                "Shortly after being shooed out of his company, he applied to fly on the Space Shuttle as a civilian astronaut (he was rejected) and even considered starting a computer company in the Soviet Union.",
+                "He actually served as a mentor for Google founders Sergey Brin and Larry Page, even sharing some of his advisers with the Google duo.",
+                "He was a pescetarian, meaning he ate no meat except for fish."
+            ]
+            
+            storedFacts["Bill Gates"] = billFacts
+            storedFacts["Steve Jobs"] = steveFacts
+        }
+    }
+    
+    func getRandomFact() -> (String, String) {
+        let person = randomPerson()
+        guard var personFacts = storedFacts[person] else{
+            return (person, "No fact found")
+        }
+        
+        let factIndex = randomIndex(fromArray: personFacts)
+        storedFacts[person]?.remove(at: factIndex)
+        
+        return (person, personFacts[factIndex])
     }
     
+    func showFact() {
+        let currentFact = getRandomFact()
+  
+        factView.text = currentFact.1
+        correctPerson = currentFact.0
+    }
     
+    func checkGuess(sender: String){
+        if sender == correctPerson{
+            count += 1
+            scoreCounter.text = "\(count)"
+            showFact()
+        } else {
+            UIView.animate(withDuration: 1, animations: {
+                self.incorrectView.alpha = 1
+            }) { _ in
+                UIView.animate(withDuration: 1, animations:{
+                    self.incorrectView.alpha = 0
+                    self.showFact()
+                })
+            }
+        }
+        
+        if count == 9{
+            let alert = UIAlertController(title: "Game Over", message: "Total correct: \(count) / 9. Would you like to play again?", preferredStyle: .alert)
+            
+            alert.addAction(UIAlertAction(title: "Yes", style: .default, handler: { action in
+                self.initialize()
+            }))
+            alert.addAction(UIAlertAction(title: "No", style: .cancel, handler: { action in
+                UIControl().sendAction(#selector(NSXPCConnection.suspend), to: UIApplication.shared, for: nil)
+            }))
+            
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
     
     
+    @IBAction func billSelect(_ sender: UIButton) {
+        if let titleLabel = sender.titleLabel {
+            if let title = titleLabel.text{
+                checkGuess(sender: title)
```

Comment:
for this one I would just do... if let title = sender.titleLabel.text {

This way you eliminated one of the if lets, always try to avoid nested if statements when possible.

Reply:
I tried that. Although both titleLabel and text returned as optionals. Hence the 2 if's. Did I miss something?